### PR TITLE
[HCCDOC#1739]: Document Insight Operator configmap config

### PR DESCRIPTION
**Pull request by kblake2016**
Added
- New module for Obfuscating DVO data using configmap
- Screenshot showing configmap formatting.

Changed
- All Insights Operator config modules using old support-secret config method.
- Nested like content together in the Using Insights Operator assembly.

Version(s): 4.11+

Issue: https://issues.redhat.com/browse/HCCDOC-1739

Link to docs preview: 
The following sections contain changes. See additional information, below for context and scope of changes.
- https://71527--ocpdocs-pr.netlify.app/openshift-enterprise/latest/support/remote_health_monitoring/insights-operator-simple-access
- https://71527--ocpdocs-pr.netlify.app/openshift-enterprise/latest/support/remote_health_monitoring/using-insights-operator#creating-insights-config-configmap_using-insights-operator
- https://71527--ocpdocs-pr.netlify.app/openshift-enterprise/latest/support/remote_health_monitoring/using-insights-operator#disabling-insights-operator-alerts_using-insights-operator
- https://71527--ocpdocs-pr.netlify.app/openshift-enterprise/latest/support/remote_health_monitoring/using-insights-operator#enabling-insights-operator-alerts_using-insights-operator
- https://71527--ocpdocs-pr.netlify.app/openshift-enterprise/latest/support/remote_health_monitoring/using-insights-operator#obfuscating-deployment-validation-operator-data_using-insights-operator
- https://71527--ocpdocs-pr.netlify.app/openshift-enterprise/latest/support/remote_health_monitoring/insights-operator-simple-access
- https://71527--ocpdocs-pr.netlify.app/openshift-enterprise/latest/support/remote_health_monitoring/insights-operator-simple-access#insights-operator-configuring-sca_remote-health-reporting-from-restricted-network

QE review:
- [Yes ] QE has approved this change.

Additional information: 
The Insights Operator previously used a support secret for operator configuration. This has been changed to ConfigMap-based configuration. I updated several procedure modules and a couple of concepts/assemblies that mentioned support secrets. Changed content consists of several procedures in two assemblies.

### CODE REVIEW

**Comment by kblake2016** [comment]
@tremes Here's the PR for the configmap changes.

File: modules/enabling-insights-operator-alerts.adoc [hunk 1]
```diff
@@ -18,14 +23,16 @@ endif::openshift-rosa,openshift-dedicated[]
...
 
 .Procedure
 
-. Navigate to *Workloads* -> *Secrets*.
-. On the *Secrets* page, select *All Projects* from the *Project* list, and then set *Show default projects* to *ON*.
-. Select the *openshift-config* project from the *Projects* list.
-. Search for the *support* secret by using the *Search by name* field.
-. Click the *Options* menu {kebab}, and then click *Edit Secret*.
-. For the `disableInsightsAlerts` key, set the *Value* field to `false`.
+. Go to *Workloads* -> *ConfigMaps* and select *Project: insights-openshift*.
```
Comment (tremes) [inline modules/enabling-insights-operator-alerts.adoc:30]: The namespace name is `openshift-insights`

File: modules/insights-operator-configuring-sca.adoc [hunk 2]
```diff
@@ -7,41 +7,29 @@
...
-... In the *Key* field, enter `scaInterval`.
-... In the *Value* field, enter `1h`.
-.. Click *Create*.
---
-+
-[NOTE]
-====
-The interval `1h` can also be entered as `60m` for 60 minutes.
-====
+. Go to *Workloads* -> *ConfigMaps* and select *Project: insights-openshift*.
```
Comment (tremes) [inline modules/insights-operator-configuring-sca.adoc:29]: Same here please.

File: modules/obfuscating-dvo-data.adoc [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+.Prerequisites
+
+* Remote health reporting is enabled, which is the default.
+* You are logged in to the {product-title} web console with the "cluster-admin" role.
+* The *insights-config* ConfigMap exists in the `insights-openshift` namespace.
+* The cluster is self managed and the Deployment Validation Operator is installed.
+
+.Procedure
+
+. Go to *Workloads* -> *ConfigMaps* and select *Project: insights-openshift*.
```
Comment (tremes) [inline modules/obfuscating-dvo-data.adoc:21]: Same here

File: modules/insights-operator-configuring-sca.adoc (hunk 2 above)
Comment (kblake2016) [reply modules/insights-operator-configuring-sca.adoc:29]: fixed

**Comment by kblake2016** [comment]
@JoeAldinger | Hi Joe. Thanks for doing the peer review for this pull request. My SME found a misnaming error in the text, which I've fixed locally. I'm waiting to send another (amended) commit until after your peer review so that I can incorporate your feedback too. If you'd prefer that I send the newest (minor) changes sooner rather than later, please let me know. Thank you.

**Comment by JoeAldinger** [comment]
I'm still commenting please wait till I submit my completed review. Thank you

**Comment by kblake2016** [comment]
Sure thing. Thanks again.

**Comment by tremes** [comment]
When looking at the labels here, this should only go to main branch and 4.15 and anywhere else.

File: modules/insights-operator-configuring-sca.adoc [hunk 4]
```diff
@@ -7,41 +7,29 @@
...
+. Click on the *insights-config* ConfigMap to open it.
+. Click *Actions* and select *Edit ConfigMap*.
+. Click the *YAML view* radio button.
+. In the file, set the *sca* attribute to `interval: 2h`; for example, to import content every two hours. You can also specify hours (h) or hours and minutes, for example: 2h30m.
+. Click *Save*.
+. On the *insights-config* ConfigMap details page, scroll down to the Data section to view `config.yaml`. You can see that the *sca* attribute is set to `interval: 2h`.
```
Comment (JoeAldinger) [inline modules/insights-operator-configuring-sca.adoc:35]:
```suggestion
. On the *insights-config* ConfigMap details page, scroll down to the Data section to view `config.yaml`. Verify that the *sca* attribute is set to `interval: 2h`.
```

File: modules/insights-operator-configuring.adoc [hunk 5]
```diff
@@ -31,15 +55,21 @@ Red Hat recommends you consult Red Hat Support before making changes to the defa
...
 
-. Navigate to *Workloads* -> *Secrets*.
-. On the *Secrets* page, select *All Projects* from the *Project* list, and then set *Show default projects* to on.
-. Select the *openshift-config* project from the *Project* list.
-. Search for the *support* secret using the *Search by name* field. If it does not exist, click *Create* -> *Key/value secret* to create it.
-. Click the *Options* menu {kebab} for the secret, and then click *Edit Secret*.
-. Click *Add Key/Value*.
-. Enter an attribute name with an appropriate value (see table above), and click *Save*.
-. Repeat the above steps for any additional configurations.
+. Go to Workloads -> ConfigMaps and in the Project drop-down menu, select openshift-insights.
```
Comment (JoeAldinger) [inline modules/insights-operator-configuring.adoc:63]: Missing *bold* for the GUI.

Comment (JoeAldinger) [inline modules/insights-operator-configuring.adoc:63]:
```suggestion
. Go to *Workloads* -> *ConfigMaps* and in the *Project* drop-down menu, select openshift-insights.
```

File: modules/insights-operator-configuring.adoc [hunk 6]
```diff
@@ -3,26 +3,50 @@
 // * support/remote_health_monitoring/using-insights-operator.adoc
 
 
-:_mod-docs-content-type: PROCEDURE
+:_mod-docs-content-type: CONCEPT
 [id="insights-operator-configuring_{context}"]
 = Configuring Insights Operator
 
-You can configure Insights Operator to meet the needs of your organization. The Insights Operator is configured using a combination of the default configurations in the `pod.yaml` file in the Insights Operator `Config` directory and the configurations stored in the `support` secret in the `openshift-config` namespace. The `support` secret does not exist by default and must be created when adding custom configurations for the first time. Configurations in the `support` secret override the defaults set in the `pod.yaml` file.
+Insights Operator configuration is a combination of the default configuration and the configuration stored in either the support secret in the `openshift-config` namespace, or the *insights-config* ConfigMap in the `openshift-insights` namespace. The ConfigMap takes precedence over the support secret.
```
Comment (JoeAldinger) [inline modules/insights-operator-configuring.adoc:10]: I'd say whatever you decide to do with the `ConfigMap` object vs config map (general usage) try to be consistent across the PR.

Comment (JoeAldinger) [inline modules/insights-operator-configuring.adoc:10]:
```suggestion
Insights Operator configuration is a combination of the default configuration and the configuration stored in either the *insights-config* ConfigMap in the `openshift-insights` namespace , or  the support secret in the `openshift-config` namespace . The ConfigMap takes precedence over the support secret.
```

File: modules/insights-operator-configuring.adoc [hunk 7]
```diff
@@ -3,26 +3,50 @@
...
+
+[NOTE]
+====
+Red Hat Insights encourages users to use the ConfigMap configuration method. Support secrets will continue to be supported in the near future but will eventually be deprecated.
```
Comment (JoeAldinger) [inline modules/insights-operator-configuring.adoc:14]: This seems like something that PX is going to need to take a look at.

Comment (JoeAldinger) [reply modules/insights-operator-configuring.adoc:14]: Try to avoid making recommendations.

File: modules/insights-operator-configuring.adoc [hunk 8]
```diff
@@ -3,26 +3,50 @@
...
+====
+
+The ConfigMap does not exist by default, so an {product-title} cluster administrator must create it. When a configuration exists, it overrides the default Operator settings that the Insights Operator reads from the `config/pod.yaml`.
```
Comment (JoeAldinger) [inline modules/insights-operator-configuring.adoc:17]:
```suggestion
The ConfigMap does not exist by default, so an {product-title} cluster administrator must create it. When a configuration exists, it overrides the default Operator settings that the Insights Operator reads from the `config/pod.yaml` directory. 
```

File: modules/insights-operator-configuring.adoc [hunk 9]
```diff
@@ -3,26 +3,50 @@
...
+
+[NOTE]
+====
+Red Hat Insights encourages users to use the ConfigMap configuration method. Support secrets will continue to be supported in the near future but will eventually be deprecated.
+====
+
+The ConfigMap does not exist by default, so an {product-title} cluster administrator must create it. When a configuration exists, it overrides the default Operator settings that the Insights Operator reads from the `config/pod.yaml`. 
+
+.Configmap configuration structure
+This example of an *insights-config* ConfigMap (config.yaml) shows configuration options using standard YAML formatting.
```
Comment (JoeAldinger) [inline modules/insights-operator-configuring.adoc:20]:
```suggestion
This example of an *insights-config* ConfigMap (`config.yaml` configuration) shows configuration options using standard YAML formatting.
```

**Review by JoeAldinger — COMMENTED** [review]
first set of comments will return to this in a bit.

**Comment by kblake2016** [comment]
@JoeAldinger ; round 1 changes added. thank you.

**Review by JoeAldinger — COMMENTED** [review]
A few more suggestions, ping me when you are ready for another review.

**Comment by kblake2016** [comment]
@JoeAldinger | Feedback incorporated.

File: modules/insights-operator-configuring.adoc [hunk 10]
```diff
@@ -31,15 +56,45 @@ Red Hat recommends you consult Red Hat Support before making changes to the defa
...
+    sca:
+      disable: false
+      interval: 2h
+    alerting:
+       disabled: false
+binaryData: {}
+immutable: false
+----
+
+. Optionally, select *Form view* and enter the necessary information that way.
```
Comment (JoeAldinger) [inline modules/insights-operator-configuring.adoc:90]:
```suggestion
. Optional: Select *Form view* and enter the necessary information that way.
```

File: modules/insights-operator-configuring.adoc [hunk 11]
```diff
@@ -31,15 +56,45 @@ Red Hat recommends you consult Red Hat Support before making changes to the defa
...
+. In the *ConfigMap Name* field, enter *insights-config*.
+. In the *Key* field, enter *config.yaml*.
+. For the *Value* field, either browse for a file to drag and drop into the field or enter your configuration parameters manually.
++
+[IMPORTANT]
+====
+Be careful to indent each parameter two spaces under its parent.
```
Comment (JoeAldinger) [inline modules/insights-operator-configuring.adoc:97]: I'm not sure this is necessary. I'd ask engr. that is just a standard rule of thumb for YAML.

File: modules/obfuscating-dvo-data.adoc [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+data:
+  config.yaml: |
+    dataReporting:
+      obfuscation:
+        - workload_names
+# ...
+----
+
+. Click *Save*. The *insights-config* config-map details page opens.
+. Verify that the value of the `config.yaml` `obfuscation` attribute is set using `- workload_names`.
```
Comment (JoeAldinger) [inline modules/obfuscating-dvo-data.adoc:41]:
```suggestion
. Verify that the value of the `config.yaml` `obfuscation` attribute is set to `- workload_names`.
```

File: modules/disabling-insights-operator-alerts.adoc [hunk 13]
```diff
@@ -23,25 +28,30 @@ endif::openshift-rosa,openshift-dedicated[]
...
 +
-* If the secret does not exist:
-.. Click *Create* -> *Key/value secret*.
-... In the *Secret name* field, enter `support`.
-... In the *Key* field, enter `disableInsightsAlerts`.
-... In the *Value* field, enter `True`.
-.. Click *Create*.
+[source,yaml]
+----
+apiVersion: v1
```
Comment (JoeAldinger) [inline modules/disabling-insights-operator-alerts.adoc:44]: I'd check that the apiVersion is correct.

**Review by JoeAldinger — COMMENTED** [review]
A few questions and small suggestion. Looks good, thanks.

**Comment by kblake2016** [comment]
@JoeAldinger, thanks for your review.

**Pull request merged**